### PR TITLE
chore: update structure of migration guide

### PR DESCRIPTION
**Pull request by erinleigh90**
#### Description of changes:
Updates structure of migration guide to allow for easier access to category sections as well as better highlighting of category-specific migration guides.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/troubleshooting/common/upgrading.mdx [hunk 1]
```diff
@@ -37,16 +37,16 @@ const handleSignIn = async ({
 };
 ```
 
-## Steps to Upgrade
+<h1>Steps to Upgrade</h1>
 
-### 1. Upgrade your dev environment
+## 1. Upgrade your dev environment
```
Comment (kvramyasri7) [inline src/fragments/lib/troubleshooting/common/upgrading.mdx:42]: super nit
```suggestion
## 1. Please update your dev environment
```

File: src/fragments/lib/troubleshooting/common/upgrading.mdx [hunk 2]
```diff
@@ -37,16 +37,16 @@ const handleSignIn = async ({
...
 
 In order to use Amplify JavaScript V6, you will need to make sure you are using the following versions in your development environment:
 
 - [Node.js](https://nodejs.org/) v16.x or later
 - [npm](https://www.npmjs.com/) v9.7.1 or later
 
-### 2. Update your Amplify project dependencies
+## 2. Upgrade your Amplify project dependencies
```
Comment (kvramyasri7) [inline src/fragments/lib/troubleshooting/common/upgrading.mdx:49]: super nit:
```suggestion
## 2. Please update your Amplify project dependencies
```

File: src/fragments/lib/troubleshooting/common/upgrading.mdx [hunk 3]
```diff
@@ -127,7 +127,7 @@ Note that v6 supports NextJS v13.5.0 through 14. We recommend upgrading if you a
 
 </BlockSwitcher>
 
-### 3. Upgrade Amplify CLI version and configuration file
+## 3. Upgrade Amplify CLI version and configuration file
```
Comment (kvramyasri7) [inline src/fragments/lib/troubleshooting/common/upgrading.mdx:130]: super nit: 
```suggestion
## 3. Please update Amplify CLI version and configuration file
```

File: src/fragments/lib/troubleshooting/common/upgrading.mdx [hunk 4]
```diff
@@ -206,15 +206,11 @@ Amplify.configure(amplifyConfig, {
 });
 ```
 
-### 4. Upgrade category usage
+## 4. Upgrade category usage
```
Comment (kvramyasri7) [inline src/fragments/lib/troubleshooting/common/upgrading.mdx:209]: super nit:
```suggestion
## 4. Please update category usage
```

File: src/fragments/lib/troubleshooting/common/upgrading.mdx [hunk 5]
```diff
@@ -524,9 +512,7 @@ async function deleteTodo() {
 
 For a deeper look at how the REST API functionality in V6, check out our [API (REST) category documentation](/[platform]/build-a-backend/restapi/).
 
-</Accordion>
-
-<Accordion title='In App Messaging' headingLevel='4'>
+### In App Messaging
```
Comment (kvramyasri7) [inline src/fragments/lib/troubleshooting/common/upgrading.mdx:515]:
```suggestion
### In-App Messaging
```

File: src/fragments/lib/troubleshooting/common/upgrading.mdx [hunk 6]
```diff
@@ -951,9 +927,13 @@ logger.error('error bar');
 
 For a deeper look at v6 Utilities, check out our [Utilities documentation](/[platform]/build-a-backend/utilities/).
 
-</Accordion>
+### Server-side Rendering
+
+<Callout>
 
-<Accordion title='Server-side Rendering' headingLevel='4'>
+DataStore is no longer supported in an SSR context: if you are using DataStore within your SSR project, you will need to migrate to the API category.
```
Comment (kvramyasri7) [inline src/fragments/lib/troubleshooting/common/upgrading.mdx:934]: Suggestion: some thing like that. Please add a link to API category as well 🙏🏼
```suggestion
We no longer support DataStore in SSR context: Please make sure to migrate to API category 
```

**Review by kvramyasri7 — CHANGES_REQUESTED** [review]

**Comment by kvramyasri7** [comment]
But why update in PR title and upgrade elsewhere?

**Review by nadetastic — DISMISSED** [review]

**Review by cwomack — APPROVED** [review]

**Review by cwomack — APPROVED** [review]

**Review by nadetastic — APPROVED** [review]

**Pull request merged**